osu.Game.Tests/Visual/SongSelect/TestSceneBeatmapCarousel.cs

```diff
@@ -112,6 +112,17 @@ public void TestTraversal()
             waitForSelection(set_count, 3);
         }
 
+        [Test]
+        public void TestNoTraversalWithoutInitialSelection()
+        {
+            // This is testing behaviour when carousel root doesn't do eager selection. This is only happens in testing.
+
+            loadBeatmaps();
+            checkNoSelection();
+            advanceSelection(direction: 1, diff: true);
+            checkNoSelection();
+        }
```

Comment:
I'm not sure it makes sense to test behaviour that only happens in the test-specific carousel class. Nobody should rely on this behaviour. I'd just probably leave an inline comment at the new version of `SelectNext()` and call it a day.

Reply:
I only left it here because if this test were to fail there would probably be others that would also fail which might make it easier to understand that having initial selection in tests is actually unwanted. But all what I said here could also be a comment somewhere.

Reply:
In that case, the tests that require null selection at the start to work should assert that before proceeding with the actual test, therefore making sure this assumption doesn't break.

Reply:
I agree with this direction. Rather than adding the new function can asserts just be added at the start of relevant tests?